sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -658,6 +677,11 @@ private void validateAuthorizationGrantType(String registrationId, AadAuthorizat
                 + AZURE_CLIENT_REGISTRATION_ID
                 + ".authorization-grant-type must be configured to 'authorization_code'.");
         }
+        if (ON_BEHALF_OF.equals(grantType)) {
+            properties.setAuthorizationGrantType(JWT_BEARER);
+            LOGGER.warn("The 'on_behalf_of' grant type is deprecated, the grant type of '{}' will be "
+                + "replaced with 'urn:ietf:params:oauth:grant-type:jwt-bearer'.", registrationId);
+        }
```

Comment:
I don't think it's the best practice to change a value in a validation method.

Reply:
Yes. This value can be an alias with another, it does not change the function.

Reply:
Will move this logic into the client registration repository.